pkg/controllers/work/apply_controller.go

```diff
@@ -458,6 +462,30 @@ func trackResourceAvailability(gvr schema.GroupVersionResource, curObj *unstruct
 	}
 }
 
+func trackCRDAvailability(curObj *unstructured.Unstructured) (ApplyAction, error) {
+	var crd apiextensionsv1.CustomResourceDefinition
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(curObj.Object, &crd); err != nil {
+		return errorApplyAction, controller.NewUnexpectedBehaviorError(err)
+	}
+	// Check if there is a condition of type "Established"
+	var establishedCondition *apiextensionsv1.CustomResourceDefinitionCondition
```

Comment:
what about "NamesAccepted" condition?

Reply:
My thought process is that if the established condition exists then the name was accepted and the CRD is either unavailable or available. That is the main condition. Otherwise, if "NamesAccepted" is true we still need to check for established and if it is false then it is unavailable (i.e. established doesn't exist).

Reply:
In this case, why the waitForCRDToBeReady waited for both conditions? I wonder if you have verified that the "NamesAccepted" condition is set to true before the "Established?

Reply:
```
// Established means that the resource has become active. A resource is established when all names are
// accepted without a conflict for the first time. A resource stays established until deleted, even during
// a later NamesAccepted due to changed names. Note that not all names can be changed.
```
https://github.com/kubernetes/kubernetes/blob/a27498ef42f27e7cddadb2c5b31bc80fed81d964/staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/types.go#L301C2-L303C88

Looks like there could be a case in which "NamesAccepted" fails after it is established and stays established. If we check for "NamesAccepted" to be true, will we consider still consider it as `manifestNotAvailableYetAction`? Technically it is still available but the name failed to be changed.